sdk/dotnet/Pulumi.Automation/Runtime/LanguageRuntimeService.cs

```diff
@@ -23,10 +24,69 @@ public LanguageRuntimeService(CallerContext callerContext)
             this._callerContext = callerContext;
         }
 
-        public override Task<GetRequiredPluginsResponse> GetRequiredPlugins(GetRequiredPluginsRequest request, ServerCallContext context)
+        public override async Task<GetRequiredPluginsResponse> GetRequiredPlugins(GetRequiredPluginsRequest request, ServerCallContext context)
         {
             var response = new GetRequiredPluginsResponse();
-            return Task.FromResult(response);
+            var assembly = this._callerContext.Program.DiscoveryAssembly;
+
+            // if discovery assembly is null, assuming consumer installed plugins manually
+            if (assembly is null)
```

Comment:
Should we make the `DiscoveryAssembly` required, and throw if it isn't present? I was thinking making it nullable would allow the consumer to disable the automatic plugin acquisition if they were installing plugins manually for whatever reason.